SkyhighSecurity/skyhigh_secure_web_gateway/ingest/parser.yml

```diff
@@ -25,27 +34,40 @@ stages:
   set_ecs_fields:
     actions:
       - set:
+          '@timestamp': '{{parse_date.datetime}}'
           event.start: '{{parse_kv.timestamp}}'
           event.category: 'network'
           event.code: '{{parse_kv.message.block_id}}'
           event.kind: 'event'
+          event.action: '{{parse_kv.message.result}}'
           event.reason: '{{parse_kv.message.block_reason}}'
-          source.ip: '{{parse_kv.message.source_ip}}'
+          source.nat.ip: '{{parse_kv.message.source_ip}}'
+          source.ip: '{{parse_kv.message.client_ip or parse_kv.message.source_ip}}'
```

Comment:
Maybe move these fields in a dedicated action with a filter on  `parse_kv.message.client_ip` like:
```yaml
      - set:
          source.nat.ip: '{{parse_kv.message.source_ip}}'
          source.ip: '{{parse_kv.message.client_ip}}'
        filter: '{{parse_kv.message.client_ip != null}}'
      - set:
          source.ip: '{{parse_kv.message.source_ip}}'
        filter: '{{parse_kv.message.client_ip == null}}
```
It will allow to not have the same value for `source.ip` and `source.nat.ip`.

Reply:
Nice improvement thank you !